servicetalk-client-api/src/main/java/io/servicetalk/client/api/ServiceDiscoverer.java

```diff
@@ -26,6 +27,10 @@
  * is called that the service discovery system will push data updates or implementations of this interface will poll for
  * data updates. Changes in the available addresses will be communicated via the resulting {@link Publisher}.
  * <p>
+ * Because typically a {@link ServiceDiscoverer} implementation runs in the background and doesn't require many compute
+ * resources, it's recommended (but not required) to run it and deliver updates on a single thread either for all
+ * discoveries or at least for all {@link Subscriber Subscribers} to the same {@link Publisher}.
```

Comment:
fwiw, it's not clear what this buys the implementors.

Reply:
I don't want to put it as a requirement, it's just a nice-to-have recommendation bcz most ppl don't think about it when they implement.
For RRLB it will help with this case: https://github.com/apple/servicetalk/pull/3005#discussion_r1676212131
But we are moving away from RRLB anyway.

In general, it makes debugging easier bcz you can track all events from a single thread and logs will represent the exact serial behavior. If they use multiple thread, serial behavior is still guaranteed by RS, but logs may print events between Subscribers in any order. 

Do you have a recommendation how to rephrase?

Reply:
My best suggestion is basically what you said in the reply. 🙂 I added as a suggestion but I'm also not strongly opinionated about it. If you prefer to keep it short and not try to sell it then that's perfectly fine with me.